src/CMakeLists.txt

```diff
@@ -0,0 +1,16 @@
+add_subdirectory(chem)
+add_subdirectory(gsi-ncdiag)
+add_subdirectory(lib-python)
+add_subdirectory(marine)
+add_subdirectory(ncep)
+add_subdirectory(ssec)
+add_subdirectory(wrfda-ncdiag)
+
+# Optional components
+if(iodaconv_gnssro_ENABLED)
+    add_subdirectory(gnssro)
+endif()
+
+if(iodaconv_bufr_ENABLED)
+  add_subdirectory(bufr)
+endif()
```

Comment:
Can we move these to the root CMakeLists.txt?  I don't see the advantages of adding another file to manage which doesn't actually define any variables or do anything other than include other files.  It just makes it more complicated to see what is being src'd.

Reply:
This keeps the top-level CMakeLists.txt clean.

Infact, the checks of enabling bufr or gnssro based on packages found or not in the top-level CMakelists belong at the level of src where those directories are going to be added.

Reply:
It's a matter of style, so whatever is prefered.  I would rather have everything I'm interested in a single file.  

The reason to not source a directory I'm not interested in is a general strategy to reduce CMake configuration overhead, which is already very significant for large bundles.  Every extra file sourced is actually multiple filesystem calls, which take forever on some systems (S4).